### PR TITLE
Collaborators improvement

### PR DESCRIPTION
**Pull request by luka-nextcloud**
Signed-off-by: Luka Trovic <luka@nextcloud.com>

* Resolves: #2449
* Target version: master 

image
image

### CODE REVIEW

File: src/components/SessionList.vue [hunk 1]
```diff
@@ -36,11 +36,12 @@
 					:disable-menu="true"
 					:show-user-status="false"
 					:disable-tooltip="true"
-					:size="44" />
+					:size="36" />
 			</div>
 		</button>
 		<template #default>
 			<div class="session-menu">
+				<p class="last-saved">Last saved: {{ lastSavedString }}</p>
```
Comment (mejo-) [inline src/components/SessionList.vue:44]: "Last saved" should probably be a translation string.

File: src/components/EditorWrapper.vue [hunk 2]
```diff
@@ -318,6 +322,10 @@ export default {
 				},
 			}
 		},
+		saveStatusClass() {
+			if (this.syncError && this.lastSavedString !== '') return 'save-error'
+			return this.dirtyStateIndicator ? 'saving-status' : 'saved-status'
+		},
```
Comment (vinicius73) [inline src/components/EditorWrapper.vue:328]: We can turn it more readable.
```suggestion
		saveStatusClass() {
			if (this.syncError && this.lastSavedString !== '') {
			  return 'save-error'
			 }
			return this.dirtyStateIndicator ? 'saving-status' : 'saved-status'
		},
```

File: src/components/SessionList.vue [hunk 3]
```diff
@@ -130,9 +137,15 @@ export default {
...
-					'background-color': session.color + ' !important',
+				if (this.showAuthorAnnotations) {
+					return {
+						border: 'solid 2px ' + session.color + ' !important',
+					}
+				} else {
+					return {
+						border: 'solid 2px var(--color-main-background) !important',
+						'background-color': session.color + ' !important',
+					}
```
Comment (vinicius73) [inline src/components/SessionList.vue:148]: We can avoid else here.

```suggestion
				if (this.showAuthorAnnotations) {
					return {
						border: 'solid 2px ' + session.color + ' !important',
					}
				}
			    return {
				  border: 'solid 2px var(--color-main-background) !important',
				  'background-color': session.color + ' !important',
			    }
```

File: src/components/SessionList.vue [hunk 4]
```diff
@@ -36,11 +36,14 @@
...
 					:disable-tooltip="true"
-					:size="44" />
+					:size="36" />
 			</div>
 		</button>
 		<template #default>
 			<div class="session-menu">
+				<p class="last-saved">
+					{{ t('text', 'Last saved') }}: {{ lastSavedString }}
+				</p>
```
Comment (vinicius73) [inline src/components/SessionList.vue:46]: We can use slot and remove the necessity of the prop `lastSavedString`.

File: src/components/SessionList.vue (hunk 3 above)
Comment (luka-nextcloud) [reply src/components/SessionList.vue:148]: Sure

**Comment by luka-nextcloud** [comment]
@vinicius73 Please check again. :)

**Review by vinicius73 — APPROVED** [review]

**Pull request merged**